lms/djangoapps/gating/tests/test_api.py

```diff
@@ -79,11 +79,11 @@ def _setup_gating_milestone(self, min_score):
         self.prereq_milestone = gating_api.get_gating_milestone(self.course.id, self.seq1.location, 'fulfills')
 
     @patch('gating.api._get_subsection_percentage')
-    @data((50, True), (100, True), (0, False))
+    @data((50, True), (100, True), (0, False), (49.50, True), (49.49, False))
     @unpack
     def test_min_score_achieved(self, module_score, result, mock_score):
         self._setup_gating_milestone(50)
-        mock_score.return_value = module_score
+        mock_score.return_value = int("{0:.0f}".format(module_score))
```

Comment:
Whats going on here is you are mocking `_get_subsection_percentage` which means the change will you made in `_get_subsection_percentage`  will not execute and mocked value will be returned.